task/move_test.go

```diff
@@ -95,3 +99,94 @@ func TestMoveCtx(t *testing.T) {
 
 	}
 }
+
+// MaybeMove tests implementation
+func (s *stub) mayBeFetch() (Thing, bool, error) {
+	if s.fetchErr == true {
+		return nil, false, errors.New("something went wrong fetching things")
+	}
+
+	if s.curr >= len(s.toFetch) {
+		return nil, false, nil
+	}
+
+	t := s.toFetch[s.curr]
+	s.curr++
+	return t, true, nil
+}
+
+func (s *stub) mayBePut(t Thing) error {
+	if s.putErr {
+		return errors.New("could not continue putting thing")
+	}
+
+	s.gotPut = append(s.gotPut, t)
+	return nil
+}
+
+func TestMaybeMove(t *testing.T) {
+	t.Run("no errors during fetch and put", func(t *testing.T) {
+		testCase := struct {
+			stub *stub
+			want []Thing
+		}{
+			stub: &stub{
+				toFetch: []Thing{1, 2, 3},
+			},
+			want: []Thing{1, 2, 3},
+		}
```

Comment:
(FYI) There is a fine line between when a case should be a separate test or part of a table. Some people deliberately separate success from failure case, and others combine them. Any logic that is more complicated than `if wantErr` should definitely never be separated.

I think it's ok to split them like you have.